### ISSUE TITLE
Missing RPM dependencies

### ISSUE THREAD

**Issue by hholst80** [shiftkey/desktop#275]
### Describe the bug

Missing RPM dependencies.

### Version & OS

Fedora Silverblue 32.

### Steps to reproduce the behavior

1. download GitHubDesktop-linux-2.4.1-linux2.rpm;
2. install `dnf install ./GitHubDesktop-linux-2.4.1-linux2.rpm`;
3. run `github-desktop`.

### Fix

Install the missing packages

    dnf install libX11-xcb nss alsa-lib

Now run `github-desktop`. Works.

Note. There is an additional dependency missing for the actual installation but I forgot to write down the command that was missing.

**Comment by shiftkey** [comment]
@hholst80 thanks for opening this up.

The list of dependencies is stored in `script/electron-builder-linux.yml`:

https://github.com/shiftkey/desktop/blob/bc37caf5dc863190d07fa8292bab7042d7caa49c/script/electron-builder-linux.yml#L13-L23

If you can figure out that last missing dependency I'd happily take a PR that adds those dependencies to the list.

**Comment by hholst80** [comment]
I have seen similar missing dependencies from VS Code as well, so those can be added as generic dependencies. Most fullsize desktop OS probably ships with these installed by default so that's why they have not been discovered until now.

**Comment by shiftkey** [comment]
@hholst80 I've incorporated some of the packaging changes in #273 and this is the list that will be part of the next release:

```
$ rpm -qpR dist/GitHubDesktop-linux-2.5.1-beta1.rpm 
(kde-cli-tools or kde-cli-tools5 or kde-runtime or trash-cli or glib2 or gvfs-client)
(libXtst or libXtst6)
(libnotify or libnotify4)
(libuuid or libuuid1)
(nss or mozilla-nss)
/bin/sh
/bin/sh
alsa-lib
at-spi2-core
gnome-keyring
gtk3
libX11-xcb
libXScrnSaver
libappindicator
libcurl
libsecret
rpmlib(CompressedFileNames) <= 3.0.4-1
rpmlib(FileDigests) <= 4.6.0-1
rpmlib(PayloadFilesHavePrefix) <= 4.0-1
rpmlib(PayloadIsXz) <= 5.2-1
rpmlib(RichDependencies) <= 4.12.0-1
xdg-utils
```

If you can spot any other dependencies that are needed, a PR to add to the `require` list in `script/package-redhat.ts` is all that we need to do in the future.

**Comment by shiftkey** [comment]
Closing this out as resolved in [`2.5.0 RC2`](https://github.com/shiftkey/desktop/releases/tag/release-2.5.0-linux2) but please open a fresh issue or submit a PR if you spot any other dependencies that were omitted.
